src/interpolation/table/Type/src/ctor.hpp

```diff
@@ -46,4 +46,19 @@ Type( Type&& other ) :
   yRange( std::move(other.yRange) ),
   searchAlgorithm( this->xRange ){}
 
+Type& operator=( const Type& other ){
+  this->xRange = other.xRange;
+  this->yRange = other.yRange;
+  this->searchAlgorithm = SearchAlgorithm{ this->xRange };
+  return *this;
+}
+
+Type& operator=( Type&& other ){
+  this->xRange = std::move(other.xRange);
```

Comment:
If you are moving data members from one instance to another, does that mean that `other` becomes unusable?

Reply:
> The post-condition is that your moved-from object is in an unspecified, but valid state.

Or in otherwords, until other is reassigned, operations on it are undefined